code-challenges.test.js

```diff
@@ -14,14 +20,36 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+
+
+describe("codeMsg", () => {
+    const secretCodeWord1 = "Lackadaisical"
+    const secretCodeWord2 = "Gobbledygook"
+    const secretCodeWord3 = "Eccentric"
+    it("takes in a string and returns a string with a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+        expect(codeMsg(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codeMsg(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codeMsg(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+//    ReferenceError: codeMsg is not defined
 
 // b) Create the function that makes the test pass.
+//create a function called codeMsg that takes a string as a parameter
+//use a built in string method .replace to return a new string with specified characters replaced. we will use .return() multiple times on the string to get each character replacement.
+//this method will only replace the first instance that it encounters. to replace  all of them, use regex with the global modifier g
+
+//     Expected: "3cc3ntr1c"
+// Received: "Ecc3ntr1c"
+
+//I forgot to account for capitals - we can use the modifier i for case insensitive matching
+
+const codeMsg = (string) => {
+return string.replace(/a/gi, "4").replace(/e/gi, "3").replace(/i/gi, "1").replace(/o/gi, "0")
```

Comment:
Keep an eye on your indentation, best practice is to have code inside of the function indented over.